D3521R0.md

```diff
@@ -200,80 +200,46 @@ Given an example like:
-    T any_cast(any&& operand);
-
-  template<class T>
-    const T* any_cast(const any* operand) noexcept;
-  template<class T>
-    T* any_cast(any* operand) noexcept;
-
-+ template<class T, same_as<any> Any>
-+   const T* try_cast(const Any& operand) noexcept;
-+ template<class T, same_as<any> Any>
-+   T* try_cast(Any& operand) noexcept;
-+ template<class T, same_as<any> Any>
-+   T* try_cast(Any&& operand) noexcept;
++ // [utility.try_cast], `try_cast`
++ template<class T>
++   requires same_as<T, remove_cvref_t<T>>
++     constexpr @*unspecified*@ try_cast = @*unspecified*@;
 ```
 
-### [any.nonmembers]{.sref} Non-member functions {.unnumbered .unlisted}
-
-```diff
-+ template<class T, same_as<any> Any>
-+   const T* try_cast(const Any& operand) noexcept;
-+ template<class T, same_as<any> Any>
-+   T* try_cast(Any& operand) noexcept;
-+ template<class T, same_as<any> Any>
-+   T* try_cast(Any&& operand) noexcept;
-```
+### Append to [utility]{.sref}:
 
 ::: add
-> *Mandates*: `is_void_v<T>` is `false`.
->
-> *Returns*: `any_cast<T>(&operand)`
-:::
+???.?.? `try_cast`                                   [utility.try_cast]
 
-### [exception.syn]{.sref} Header `<exception>` synopsis {.unnumbered .unlisted}
+[1]{.pnum} The name `try_cast` denotes a customization point object ([customization.point.object]).
 
-```diff
-  exception_ptr current_exception() noexcept;
-  [[noreturn]] void rethrow_exception(exception_ptr p);
-  template <class E>
-    const E* exception_ptr_cast(const exception_ptr& p) noexcept;
-  template <class T> [[noreturn]] void throw_with_nested(T&& t);
-
-+ template<class E>
-+   const E* try_cast(const exception_ptr& p) noexcept;
-```
+[2]{.pnum} Given a subexpression `E` with type `U`, let `u` be an lvalue that denotes the reified object for `E`. Then:
 
-### [propagation]{.sref} Exception propagation {.unnumbered .unlisted}
+ - [2.1]{.pnum} If `E` is an rvalue, `try_cast(E)` is ill-formed.
 
-```diff
-+ template<class E>
-+   const E* try_cast(const exception_ptr& p) noexcept;
-```
+ - [2.2]{.pnum} Otherwise, if `same_as<remove_cvref_t<U>, any>` is `true` ([any]), then `try_cast(E)` is expression-equivalent to `any_cast<T>(&u)` if that is well-formed, and is ill-formed otherwise.
 
-::: add
-> *Mandates*: `E` is a *cv*-unqualified complete object type. `E` is not an array type.
-> `E` is not a pointer or pointer-to-member type.
->
-> *Returns*: `exception_ptr_cast<E>(p)`
+ - [2.3]{.pnum} Otherwise, if `same_as<remove_cvref_t<U>, exception_ptr>` is `true` ([propagation]), then `try_cast(E)` is expression-equivalent to `exception_ptr_cast<T>(&u)` if that is well-formed, and is ill-formed otherwise.
+
+ - [2.4]{.pnum} Otherwise, if `U` is a class type and `auto(try_cast(u))` is a valid expression whose type is `T *` or `const T *`, where the meaning of `try_cast` is established as-if by performing argument-dependent lookup only ([basic.lookup.argdep]), then `try_cast(E)` is expression-equivalent to that expression.
```

Comment:
Pass `u` or `&u`? Feels like should be `&u`

Reply:
Huh.  Maybe.  That sounds like a poll.

Reply:
Nah, you're right.  As soon as I started writing out the two options, it was obvious it should just be consistent with `any`.